codec-http/src/main/java/io/netty5/handler/codec/http/HttpContentDecoder.java

```diff
@@ -50,106 +51,117 @@ public abstract class HttpContentDecoder extends MessageToMessageDecoder<HttpObj
-            if (msg instanceof LastHttpContent) {
-                continueResponse = false;
+            if (continueResponse) {
+                if (msg instanceof LastHttpContent) {
+                    continueResponse = false;
+                }
+                // 100-continue response must be passed through.
+                fireChannelRead(ctx, msg);
+                return;
             }
-            // 100-continue response must be passed through.
-            fireChannelRead(ctx, msg);
-            return;
-        }
 
-        if (msg instanceof HttpMessage) {
-            cleanup();
-            final HttpMessage message = (HttpMessage) msg;
-            final HttpHeaders headers = message.headers();
+            if (msg instanceof HttpMessage) {
+                cleanup();
+                final HttpMessage message = (HttpMessage) msg;
+                final HttpHeaders headers = message.headers();
 
-            // Determine the content encoding.
-            String contentEncoding = headers.get(HttpHeaderNames.CONTENT_ENCODING);
-            if (contentEncoding != null) {
-                contentEncoding = contentEncoding.trim();
-            } else {
-                String transferEncoding = headers.get(HttpHeaderNames.TRANSFER_ENCODING);
-                if (transferEncoding != null) {
-                    int idx = transferEncoding.indexOf(",");
-                    if (idx != -1) {
-                        contentEncoding = transferEncoding.substring(0, idx).trim();
+                // Determine the content encoding.
+                String contentEncoding = headers.get(HttpHeaderNames.CONTENT_ENCODING);
+                if (contentEncoding != null) {
+                    contentEncoding = contentEncoding.trim();
+                } else {
+                    String transferEncoding = headers.get(HttpHeaderNames.TRANSFER_ENCODING);
+                    if (transferEncoding != null) {
+                        int idx = transferEncoding.indexOf(",");
+                        if (idx != -1) {
+                            contentEncoding = transferEncoding.substring(0, idx).trim();
+                        } else {
+                            contentEncoding = transferEncoding.trim();
+                        }
                     } else {
-                        contentEncoding = transferEncoding.trim();
+                        contentEncoding = IDENTITY;
                     }
-                } else {
-                    contentEncoding = IDENTITY;
                 }
-            }
-            decompressor = newContentDecoder(contentEncoding);
+                decompressor = newContentDecoder(contentEncoding);
 
-            if (decompressor == null) {
-                fireChannelRead(ctx, message);
-                return;
-            }
+                if (decompressor == null) {
+                    if (message instanceof HttpContent) {
+                        dispose = false;
+                    }
```

Comment:
Are there actually any cases where we _do_ want to dispose here? I _think_ we could unconditionally set `dispose = false`

Reply:
I agree, let me do the change